onnxruntime/core/providers/webgpu/math/cum_sum.cc

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/webgpu/math/cum_sum.h"
+#include "core/providers/webgpu/shader_helper.h"
+#include "core/providers/webgpu/webgpu_supported_types.h"
+
+namespace onnxruntime {
+namespace webgpu {
+
+ONNX_OPERATOR_VERSIONED_KERNEL_EX(
+    CumSum,
+    kOnnxDomain,
+    11, 13,
+    kWebGpuExecutionProvider,
+    (*KernelDefBuilder::Create()).TypeConstraint("T", WebGpuSupportedFloatTypes()).TypeConstraint("T2", WebGpuSupportedNumberTypes()).InputMemoryType(OrtMemTypeCPU, 1),
+    CumSum);
+
+ONNX_OPERATOR_KERNEL_EX(
+    CumSum,
+    kOnnxDomain,
+    14,
+    kWebGpuExecutionProvider,
+    (*KernelDefBuilder::Create()).TypeConstraint("T", WebGpuSupportedFloatTypes()).TypeConstraint("T2", WebGpuSupportedNumberTypes()).InputMemoryType(OrtMemTypeCPU, 1),
+    CumSum);
+
+Status CumSumProgram::GenerateShaderCode(ShaderHelper& shader) const {
+  const ShaderVariableHelper& input = shader.AddInput("input", ShaderUsage::UseUniform);
+  const ShaderVariableHelper& output = shader.AddOutput("output", ShaderUsage::UseUniform);
+
+  std::string index = "i32(" + input.IndicesGet("input_indices", "uniforms.axis") + ")";
+  std::string max = GetElementAt("uniforms.input_shape", "uniforms.axis", input.Rank());
+  std::stringstream lowerLimit;
+  lowerLimit << "let first : i32 = 0;\n"
+             << "if (uniforms.reverse == 1) {\n"
+             << "  first = " << index << ";\n"
+             << "  if (uniforms.exclusive == 1) { first += 1; }\n"
+             << "}\n";
+  std::stringstream upperLimit;
+  upperLimit << "let last : i32 = 0;\n"
+             << "if (uniforms.reverse == 1) {\n"
+             << "  last = " << max << ";\n"
+             << "} else {\n"
+             << "  last = " << index << ";\n"
+             << "  if (uniforms.exclusive == 0) { last += 1; }\n"
+             << "}\n";
```

Comment:
`shader.MainFunctionBody()` is already a stream. suggest to follow the stream instead of creating multiple ones.